### PR TITLE
DIV-4672 amend family man keep

### PR DESCRIPTION
**Pull request by alexwakeman**
[Amend Case: Family Man Reference - System should not allow to change FMR after Amend and issue the case in CCD](https://tools.hmcts.net/jira/browse/DIV-4672)

We need to keep the D8 Case Reference for the Amend Petition draft

### CODE REVIEW

File: charts/div-cms/values.aat.template.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ java:
         DRAFT_STORE_API_BASEURL : "http://draft-store-service-aat.service.core-compute-aat.internal"
         IDAM_CASEWORKER_USERNAME : "DivCaseWorker@AAT.com"
         APPLICATION_INSIGHTS_IKEY : "63c30ef9-a27b-4d38-8afc-4c180fb6ac33"
-        CASE_FORMATTER_SERVICE_API_BASEURL : "http://div-cfs-aat.service.core-compute-aat.internal"
+        CASE_FORMATTER_SERVICE_API_BASEURL : "http://div-cfs-pr-123.service.core-compute-preview.internal"
```
Comment (tchow8) [inline charts/div-cms/values.aat.template.yaml]: Probably don't need this anymore?

File: .gitignore [hunk 2]
```diff
@@ -23,3 +23,5 @@ java-rpm-packaging/
 application-*.yml
 application-*.yaml
 application-local.properties
+
+settings-aat.json
```
Comment (hosainnet) [inline .gitignore]: This is just a randomly named file name that doesn't follow the convention, if you use `application-aat.yaml` you won't need this :P

File: charts/div-cms/values.aat.template.yaml (hunk 1 above)
Comment (alexwakeman) [reply charts/div-cms/values.aat.template.yaml]: We do for testing for now

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java [hunk 3]
```diff
@@ -105,6 +105,13 @@ private void createCaseWorkerCourtAdminUserInIdam(String username, String emailA
                 .build();
 
         idamUtils.createUserInIdam(registerUserRequest);
+
+        try {
+            //give the user some time to warm up..
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (hosainnet) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java]: Probably shouldn't have stacktrace..

File: .gitignore (hunk 2 above)
Comment (hosainnet) [reply .gitignore]: See
https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-profiles.html
And
https://stackoverflow.com/questions/38520638/how-to-set-spring-profile-from-system-variable

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java (hunk 3 above)
Comment (alexwakeman) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java]: Should remove from COS then I guess..

Comment (alexwakeman) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java]: Actually, this is only for testing and could be useful debugging issues with IDAM

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/CcdSubmissionSupport.java [hunk 4]
```diff
@@ -50,8 +50,11 @@ protected Response submitCaseJson(String jsonCase, String userToken) {
 
     String loadJson(String fileName, UserDetails userDetails) throws Exception {
         // Update document links in the Json String to be current environment
-        return loadJson(fileName, PAYLOAD_CONTEXT_PATH)
-            .replaceAll("-aat", "-".concat(testEnvironment))
+        String payload = loadJson(fileName, PAYLOAD_CONTEXT_PATH);
+        if (!testEnvironment.equals("local")) {
```
Comment (alexwakeman) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/CcdSubmissionSupport.java]: Needed to ensure tests pass locally. URL re-write does not apply locally.

**Comment by nsingamsetty** [comment]
Testing completed.All looking good. Refer DIV-4672 for additional evidences and comments.

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java (hunk 3 above)
Comment (hosainnet) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/IdamTestSupport.java]: Already removed last week
https://github.com/hmcts/div-case-orchestration-service/blob/cd1ada914f8265a8012ad16745e30f04100e9fd0/src/integrationTest/java/uk/gov/hmcts/reform/divorce/context/IntegrationTest.java#L83

**Pull request merged**